html/indexbase.html

```diff
@@ -220,8 +226,50 @@ <h3 class="panel-title">Legend</h3>
             trigger: "hover",
             placement: "auto"
         })
+    }
+}
 
+var wereRacksUpdated = false;
+function updateRacksOnce(statusObject) {
+    if (wereRacksUpdated) {
+        return;
+    }
+    var nrRacks = 0;
+    var nrHostsPerRow = 2;
+    var racks = [];
+    var racksThatHaveHosts = new Set();
+    for (var i in statusObject.hosts) {
+        var hostObject = statusObject.hosts[i];
+        var rackNumber = parseInt(hostObject.id.replace("rack", "").replace("-server\d", ""));
+        if (rackNumber > nrRacks) {
+            nrRacks = rackNumber;
+        }
+        racksThatHaveHosts.add(rackNumber);
+        var serverName = hostObject.id.replace(/rack\d+-server/, "");
```

Comment:
use `id.match((/rack\d+-server(\d+)/)` this better conveys what you are doing here. Also `serverName` is a bad name and not actually needed, you can inline it